File: frontend/src/services/adminDashboard.ts
"use client";
import { apiFetch } from "@/services/api";

export type AdminDashboardStats = {
  users: {
    total: number;
    owners: number;
    clients: number;
    active: number;
    inactive: number;
  };
  properties: {
    total: number;
    pending: number;
    approved: number;
    rejected: number;
  };
  bookings: {
    total: number;
    pending: number;
    confirmed: number;
    cancelled: number;
    completed: number;
  };
  financials: {
    total_revenue: number | string;
    platform_commission: number | string;
    owner_earnings: number | string;
    outstanding_cash_commission: number | string;
  };
  reviews: {
    total: number;
    average_rating: number | string;
  };
};

export async function getAdminDashboardStats(): Promise<AdminDashboardStats> {
  return apiFetch("/admin/dashboard/stats");
}

File: backend/app/schemas/admin_dashboard.py
from decimal import Decimal

from pydantic import BaseModel


class AdminDashboardUserStats(BaseModel):
    total: int
    owners: int
    clients: int
    active: int
    inactive: int


class AdminDashboardPropertyStats(BaseModel):
    total: int
    pending: int
    approved: int
    rejected: int


class AdminDashboardBookingStats(BaseModel):
    total: int
    pending: int
    confirmed: int
    cancelled: int
    completed: int


class AdminDashboardFinancialStats(BaseModel):
    total_revenue: Decimal
    platform_commission: Decimal
    owner_earnings: Decimal
    outstanding_cash_commission: Decimal


class AdminDashboardReviewStats(BaseModel):
    total: int
    average_rating: Decimal


class AdminDashboardStatsResponse(BaseModel):
    users: AdminDashboardUserStats
    properties: AdminDashboardPropertyStats
    bookings: AdminDashboardBookingStats
    financials: AdminDashboardFinancialStats
    reviews: AdminDashboardReviewStats